### ISSUE TITLE
Search view Textfield becomes unresponsive

### ISSUE THREAD

**Issue by JordanADavies** [flutter/flutter#23862]
## Steps to Reproduce

1. Navigate to search screen.
2. Enter search term and tap on a result.
3. Navigate away from search screen.
4. Return to search screen using the back button.
5. Search text field becomes unresponsive.

Reproduced here: https://gist.github.com/JordanADavies/c7b630681aa81940d254876078a2f7a0
```
flutter doctor -v
[✓] Flutter (Channel beta, v0.9.4, on Mac OS X 10.14.1 18B75, locale en-GB)
    • Flutter version 0.9.4 at /Users/jordandavies/Documents/development/flutter
    • Framework revision f37c235c32 (5 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/jordandavies/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 2.0.0
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 29.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Connected devices (1 available)
    • iPhone XR • 4018CC7F-D913-4181-8CF3-C29A87529EA2 • ios • iOS 12.1 (simulator)

• No issues found!
```

**Comment by timeuser** [comment]
Same problem here. You can clear the search field by tapping the 'x' but still can't enter any new search query. Seems to be no way to focus the search field again to type a new query.

**Comment by sergiuiacob1** [comment]
Facing the same issue. Is there any workaround for this?

**Comment by zoechi** [comment]
Similar to #19750

**Comment by edmundackah** [comment]
Any fix for this issue yet?

**Comment by amirfeqhi** [comment]
Any fix yet?

**Comment by Luk3s** [comment]
If this only happens with `SearchDelegate` (which is how I ended up here) can we clarify it in the title of the issue and in the description of the fist comment? I can only get to this issue by checking other closed issues, not very search friendly.

If it happens in other occasions, can we add more information? I feel this issue needs more love to be solved.

**Comment by Luk3s** [comment]
I have tried recreating the `SearchDelegate` and I noticed that when the `maintainState` flag of the `PageRoute` is set to false (`SearchDelegate` uses its own page route called `_SearchPageRoute`) then we have the problems with the unresponsive text field.

**Comment by scottnimos** [comment]
I have also encountered this issue

**Comment by zoechi** [comment]
@scottnimos Using "add reaction" on the initial comment would increase priority while `+1`/`me too` comments are rather pointless and cause lots of people to get notified for no reason.
To get notified yourself use the [Subscribe] button to the top right instead.

**Comment by cahsul** [comment]
I have the same problem

**Comment by zoechi** [comment]
@scottnimos @cahsul Using "add reaction" on the initial comment would increase priority while `+1` comments are rather pointless and cause lots of people to get notified for no reason.
To get notified yourself use the [Subscribe] button to the top right instead.

**Comment by zoechi** [comment]
Might be related to #14792, #26810, #25933

**Comment by HansMuller** [comment]
This issue was fixed in #27365 (sorry that it's taken so long)
